thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js

```diff
@@ -236,33 +231,22 @@ export default Controller.extend({
    * @param {Object} newId - conditional param to help us tag any function that was "just added"
    * @return {RSVP.Promise} A new list of functions (alerts)
    */
-  prepareFunctions(configGroup, newId = 0) {
+  prepareFunctions(configGroup) {
+    const existingFunctionList = configGroup.emailConfig ? configGroup.emailConfig.functionIds : [];
     const newFunctionList = [];
-    const existingFunctionList = _.has(configGroup, 'emailConfig') ? configGroup.emailConfig.functionIds : [];
     let cnt = 0;
 
     // Build object for each function(alert) to display in results table
     return new RSVP.Promise((resolve) => {
-      existingFunctionList.forEach((functionId) => {
+      for (var functionId of existingFunctionList) {
```

Comment:
Is there a reason why this was changed from a `forEach` to a `for` loop? Is `existingFunctionList` ever an object? It seems to me we can still keep `forEach` since it's more readable.

Reply:
Yeah - it got reverted. Correcting.